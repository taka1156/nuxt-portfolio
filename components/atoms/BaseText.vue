<template functional>
  <p class="base-text" :class="data.staticClass || ''">
    <slot />
  </p>
</template>

<style scoped>
/* css reset */
p {
  margin: 0;
  padding: 0;
}

/* css reset */

.base-text {
  padding: 5px;
}

.base-text--card {
  display: block;
  height: 100px;
  width: 100%;
  margin: 0 auto;
  text-align: left;
  word-wrap: break-word;
  box-sizing: border-box;
}

.base-text--nav {
  display: inline;
  color: white;
}
</style>
