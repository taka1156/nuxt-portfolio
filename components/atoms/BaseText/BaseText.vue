<template>
  <p class="base-text--extend">
    <!-- @slot テキスト -->
    <slot />
  </p>
</template>

<script>
export default {
  name: 'BaseText'
};
</script>

<style scoped>
/* css reset */
p {
  padding: 0;
  margin: 0;
}

/* css reset */
</style>
