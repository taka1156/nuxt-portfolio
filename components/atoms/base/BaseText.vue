<template>
  <p class="base-text" :class="`base-text--${useType}`">
    <!-- @slot テキスト -->
    <slot />
  </p>
</template>

<script>
export default {
  name: 'BaseText',
  props: {
    useType: {
      type: String,
      default: 'none',
      validator: function (value) {
        return ['none', 'nav', 'card'].indexOf(value) !== -1;
      },
    },
  },
};
</script>

<style scoped>
/* css reset */
p {
  margin: 0;
  padding: 0;
}

/* css reset */

.base-text {
  padding: 5px;
}

.base-text--card {
  display: block;
  height: 100px;
  width: 100%;
  margin: 0 auto;
  text-align: left;
  word-wrap: break-word;
  box-sizing: border-box;
}

.base-text--nav {
  display: inline;
  color: white;
}
</style>
