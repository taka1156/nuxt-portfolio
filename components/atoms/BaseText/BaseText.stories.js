import BaseText from './BaseText.vue';

export default {
  title: 'Atoms/BaseText',
  component: BaseText
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { BaseText },
  template: '<base-text v-bind="$props">Text</base-text>'
});

export const Default = Template.bind({});
Default.parameters = {
  backgrounds: {
    default: 'dark'
  }
};
