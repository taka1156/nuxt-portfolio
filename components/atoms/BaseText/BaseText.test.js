import { mount } from '@vue/test-utils';
import BaseText from './BaseText.vue';

describe('BaseText', () => {
  const dummyText = 'ダミーテキスト';
  const baseText = () =>
    mount(BaseText, {
      slots: {
        default: dummyText
      }
    });

  it('値がDOMに反映されているか', () => {
    const wrapper = baseText();
    expect(wrapper.text()).toBe(dummyText);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
