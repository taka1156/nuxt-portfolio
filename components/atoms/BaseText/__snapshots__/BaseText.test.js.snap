// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseText 値がDOMに反映されているか 1`] = `"<p class=\\"base-text--extend\\">ダミーテキスト</p>"`;
